lib/rpc/rpc_sidechain.dart

```diff
@@ -179,14 +229,44 @@ class SidechainRPCLive implements SidechainRPC {
 
   @override
   Future<int> mainchainBlockCount() async {
-    final cached = await _client.call('updatemainblockcache') as Map<String, dynamic>;
+    final cached = await _client?.call('updatemainblockcache') as Map<String, dynamic>;
 
     return cached['cachesize'];
   }
 
   @override
   Future<int> blockCount() async {
-    return await _client.call('getblockcount');
+    return await _client?.call('getblockcount');
+  }
+
+  @override
+  Future<(bool, String?)> testConnection() async {
+    try {
+      final pong = await _client?.call('ping') as Map<String, dynamic>?;
+      if (pong == null) {
+        log.i('pong was null, setting connected to true');
+        connectionError = null;
+        connected = true;
+      } else {
+        log.i('connection error was set!');
+        connectionError = pong['error'];
+        connected = false;
+      }
+    } catch (error) {
+      log.i('connection error threw error ${error.toString()}!');
+      connectionError = error.toString();
+      connected = false;
+    }
+
+    notifyListeners();
+
+    return (connected, connectionError);
+  }
```

Comment:
same here as above. could this somehow be extracted to the `RPCConnection` abstract class? Proper OOP style 🤓

Reply:
O.O

Reply:
it sure can! nice